sdk/python/generator/Generator.py

```diff
@@ -71,39 +58,33 @@ def to_type_formatter_instance(type_instance):
 	return type_formatter(type_instance)
 
 
-def to_virtual_type_instance(yaml_descriptor):
+def to_virtual_type_instance(ast_model):
 	object_type = None
-	if yaml_descriptor['type'] == 'struct':
+	if isinstance(ast_model, Struct):
 		object_type = StructObject
-	elif yaml_descriptor['type'] == 'enum':
+	elif isinstance(ast_model, Enum):
 		object_type = EnumObject
-	elif yaml_descriptor['type'] == 'byte':
-		size = yaml_descriptor['size']
-		# can be either 'bound' array or normal
-		if isinstance(size, str):
-			object_type = ArrayObject
-		else:
-			object_type = ArrayObject if size > 8 else IntObject
+	elif isinstance_builtin(ast_model, FixedSizeBuffer):
+		object_type = ArrayObject
+	elif isinstance_builtin(ast_model, FixedSizeInteger):
+		object_type = IntObject
 
-	return object_type(yaml_descriptor)
+	return object_type(ast_model)
 
 
 def create_printer(type_instance, name):
-	if type_instance.is_builtin:
+	if type_instance.is_builtin or isinstance(type_instance.ast_model, Alias):
 		return BuiltinPrinter(type_instance, name)
 
 	return create_pod_printer(type_instance, name)
 
 
 def process_concrete_struct(type_map, struct_type, abstract_impl_map):
-	factory_type = type_map[struct_type.yaml_descriptor['factory_type']]
+	factory_type = type_map[struct_type.ast_model.factory_type]
 
-	discriminators = factory_type.yaml_descriptor.get('discriminator', [])
+	discriminators = [] if not hasattr(factory_type, 'discriminator') else factory_type.discriminator
 	for i, discriminator in enumerate(discriminators):
-		discriminators[i] = fix_name(discriminator)
-
-	for init in factory_type.yaml_descriptor['initializers']:
-		fix_item(init, 'target_property_name')
```

Comment:
hmm, so where/how are you doing this? (I mean this is needed at a later stage when comparing against "already fixed" field name iirc)

Reply:
ok, I assume everything is moved to printers stage, which seems like a sane thing to do

Reply:
> ok, I assume everything is moved to printers stage, which seems like a sane thing to do

yep. that made it easier for me to follow.